### DISCUSSION TITLE
how to limit search range, but allow searches in future

### DISCUSSION THREAD

**Discussion by textual** [gpbl/react-day-picker#2515] [Support]
i have an event database and i don't want someone to just search for an entire year. but I would allow them to go forward a few months and search for something in the future. i set my max days to 30, and the undesired side effect is that I need to straddle dates to go into the future more than 30 days. it wont allow me to just go forward, I need to move the start date forward, then move the end date, and repeat until I can get a month or two into the future. is there a better way to handle this, like maybe clear the dates to allow them to set a start date whenever, and then enforce the 30 days max thing?

Im using the date range sample from shadcn, and I'm setting a bunch of vars. I'm not sure how to simulate my next js project in codepen, but here's the calendar component which is wrapped in a popover content component

```
//example js
const date = {to:new Date(), from = new Date().addDays(15)};

            <Calendar
              initialFocus
              mode="range"
              defaultMonth={date?.from}
              selected={date}
              onSelect={setDate}
              numberOfMonths={2}
              max={30}
              // minDate={new Date()}
              disabled={{ before: new Date() }}
            />
```

**Comment by wilsonadenuga** [comment]
@textual maybe you can share a code sample on codesandbox with a brief explanation of how it should work, then I can take a look at the code and suggest solutions

**Reply by textual** [reply]
not sure how to setup code sandbox to mimic my project, updated op instead

**Comment by gpbl** [comment]
Maybe @textual is trying to change `startMonth` or `endMonth` dynamically? In that case, the fix might be to control the month displayed in the calendar based on `startMonth`.

```tsx
function Controlled() {
  const [month, setMonth] = useState(new Date());

  return (
    <DayPicker month={month} onMonthChange={setMonth} startMonth={variableStartMonth} />
  );
}
```

More code samples could help clarify your issue, @textual.

**Reply by textual** [reply]
i updated my op with code sample, if that helps
